File: CF PABC.NET/I.pas
begin
  var N := ReadlnInteger();
  
  var X := new List<integer>;
  var Y := new List<Integer>;
  
  loop N do
  begin
    var (A, B) := ReadlnInteger2();
    X.Add(A);
    Y.Add(B);
  end;
  
  var (XMean, YMean) := (X.Average, Y.Average);
  var Numerator := X.Zip(Y, (A, B) -> (A - XMean) * (B - YMean)).Sum();
  var Denominator := (X.Select(A -> (A - XMean).Sqr).Sum() * Y.Select(A -> (A - YMean).Sqr).Sum()).Sqrt;
  var Ro := Denominator <> 0 ? Numerator / Denominator : 0;
  Writeln(Ro:0:9);
end.

File: CF Python/I.py
n = int(input())

X = []
Y = []

for i in range(n):
    x, y = map(int, input().split())
    X.append(x)
    Y.append(y)

x_mean = sum(X) / len(X)
y_mean = sum(Y) / len(Y)

numerator = sum([(x - x_mean) * (y - y_mean) for x, y in zip(X, Y)])
denominator = (sum([(x - x_mean) ** 2 for x in X]) * sum([(y - y_mean) ** 2 for y in Y])) ** 0.5
ro = numerator / denominator if denominator != 0 else 0

print(f'{ro:.9f}')
